src/utils/init/config-github.js

```diff
@@ -1,188 +1,215 @@
-      opts: {
-        scopes: ['admin:org', 'admin:public_key', 'repo', 'user'],
-        userAgent: UA,
-        note: `Netlify CLI ${os.userInfo().username}@${os.hostname()}`,
-      },
-      log: ctx.log,
-    })
-    globalConfig.set(`users.${current}.auth.github`, newToken)
-    ghtoken = newToken
+  if (isValidToken(githubToken)) {
+    return githubToken.token
   }
-  const octokit = new Octokit({
-    auth: `token ${ghtoken.token}`,
-  })
-
-  const key = await api.createDeployKey()
-  const parsedURL = parseGitRemote(repo.repo_path)
 
-  await octokit.repos.addDeployKey({
-    title: 'Netlify Deploy Key',
-    key: key.public_key,
-    repo: parsedURL.name,
-    owner: parsedURL.owner,
-    read_only: true,
+  const newToken = await ghauth({
+    log,
   })
+  globalConfig.set(`users.${userId}.auth.github`, newToken)
+  return newToken.token
+}
 
-  repo.deploy_key_id = key.id
-
-  // TODO: Look these up and default to the lookup order
+const getGitHubClient = ({ token }) => {
+  const octokit = new Octokit({
+    auth: `token ${token}`,
+  })
+  return octokit
+}
 
-  let defaultBuildCmd
-  let defaultBuildDir = '.'
-  // read from netlify toml
-  const { build } = ctx.netlify.config
-  if (build && build.command) {
-    defaultBuildCmd = build.command
-  }
-  if (build && build.publish) {
-    defaultBuildDir = build.publish
+const addDeployKey = async ({ log, api, octokit, repoOwner, repoName, failAndExit }) => {
+  log('Adding deploy key to repository...')
+  const key = await createDeployKey({ api, failAndExit })
+  try {
+    await octokit.repos.addDeployKey({
+      title: 'Netlify Deploy Key',
+      key: key.public_key,
+      owner: repoOwner,
+      repo: repoName,
+      read_only: true,
+    })
+    log('Deploy key added!')
+    return key
+  } catch (error) {
+    let message = formatErrorMessage({ message: 'Failed adding GitHub deploy key', error })
+    if (error.status === 404) {
+      const { name, owner } = formatRepoAndOwner({ repoName, repoOwner })
+      message = `${message}. Does the repository ${name} exist and do ${owner} has the correct permissions to set up deploy keys?`
+    }
+    failAndExit(message)
   }
-  if (build && build.functions) console.log(`Netlify functions folder is ${chalk.yellow(build.functions)}`)
-  const { buildCmd, buildDir } = await inquirer.prompt([
-    {
-      type: 'input',
-      name: 'buildCmd',
-      message: 'Your build command (hugo build/yarn run build/etc):',
-      filter: (val) => (val === '' ? '# no build command' : val),
-      default: defaultBuildCmd,
-    },
-    {
-      type: 'input',
-      name: 'buildDir',
-      message: 'Directory to deploy (blank for current dir):',
-      default: defaultBuildDir,
-    },
-  ])
+}
 
-  const tomlpath = path.join(ctx.netlify.site.root, 'netlify.toml')
-  const tomlDoesNotExist = !fs.existsSync(tomlpath)
-  if (tomlDoesNotExist && (!ctx.netlify.config || Object.keys(ctx.netlify.config).length === 0)) {
```

Comment:
This code was moved to `saveNetlifyToml`.
Also there is a bug here as `ctx.netlify.config` is never undefined or empty:
https://github.com/netlify/build/blob/60ee6cfb5af57e4fed70fce7bc566d086ce4d653/packages/config/src/files.js#L18